tests/kernel: timer_api: Enable Test userspace

erwango opened the pull request:
timer_api requires TEST_USERSPACE activation which is missing
in tickless configuration of the test.
Enable flag in prj_tickless.cnf

Fixes #20904


Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

ioannisg reviewed (commented):
Thanks for reporting this issue @erwango.
I see that the .tickless test-case is supposed to be build-only.
I wonder, now if this test-case is intentionally supposed to be without user-mode. I don't know the history here. Perhaps @andrewboie can help.

Also, I notice that TICKLESS is set to "y", which seems redundant to me, right? It's default y in kernel/Kconfig.

So, I am not sure what the purpose of the test is. Also I notice that is sets SYS_POWER_MANAGEMENT (not present in prj.conf), but this is not reflected in the test-case name.

Otherwise, if we, indeed want to make the test work with user mode, the fix is fine. But then, why "build-only)?

erwango commented:
> Thanks for reporting this issue @erwango.
> I see that the .tickless test-case is supposed to be build-only.

From what I saw, it was made build only because failing in CI (cf https://github.com/zephyrproject-rtos/zephyr/pull/13994).
It is running fine with this fix, so I've pushed https://github.com/zephyrproject-rtos/zephyr/pull/20907 to see what's happening.

erwango committed:
tests/kernel: timer_api: Enable Test userspace

timer_api requires TEST_USERSPACE activation which is missing
in tickless configuration of the test.
Enable flag in prj_tickless.cnf

Fixes #20904


Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

ioannisg commented:
Are you going to keep the build-only flag?
